packages/web-react/src/components/FileUploader/useFileUploaderInput.ts

```diff
@@ -0,0 +1,154 @@
+import { RefObject, ChangeEvent, DragEvent, useEffect, useState } from 'react';
+import { useFileUploaderContext } from './FileUploaderContext';
+import { FileUploaderQueueItemProps } from '../../types';
+
+export interface UseFileUploaderInputProps {
+  dropZoneReference: RefObject<HTMLDivElement>;
+  fileLimitMaxSize: number;
+  fileQueueLimit: number;
+  hasDragAndDrop: boolean;
+  inputReference: RefObject<HTMLInputElement>;
+  onError?: (error: string) => void;
+  fileQueue2: FileUploaderQueueItemProps[]; // TODO: to sem normálně nepatří
+}
+
+export interface UseFileUploaderInputReturn {
+  isDragging: boolean;
+}
+
+export const useFileUploaderInput = (props: UseFileUploaderInputProps): UseFileUploaderInputReturn => {
+  const { hasDragAndDrop, inputReference, dropZoneReference, fileLimitMaxSize, fileQueueLimit, fileQueue2, onError } =
+    props;
+
+  const [isDragging, setDragging] = useState(false);
+
+  const { fileQueue, addToQueue } = useFileUploaderContext();
+  console.log('on init', fileQueue, fileQueue2);
+  const getUpdatedFileName = (name: string): string => `file__${name.replace(/\./g, '_')}`;
+
+  const checkAllowedFileSize = (file: File) => {
+    if (file.size > fileLimitMaxSize) {
+      throw new Error('The file size limit has been exceeded');
+    }
+  };
+
+  const checkFileDuplicity = (file: File) => {
+    // TODO: This wont work !!!
+    console.log('checkFileDuplicity', fileQueue, fileQueue2);
+    const index = fileQueue.findIndex((item) => item.file === file);
+    if (index > -1) {
+      throw new Error('This file already exists in the queue');
+    }
+  };
+
+  const checkQueueSize = () => {
+    // TODO: This wont work !!!
+    console.log('checkQueueSize', fileQueue.length, fileQueue2.length);
+    if (fileQueue.length >= fileQueueLimit) {
+      throw new Error('You have exceeded the number of files allowed in the queue');
+    }
+  };
+
+  const fileProcessHandler = (file: File) => {
+    console.log('fileProcessHandler', fileQueue.length, fileQueue2.length);
+    try {
+      checkAllowedFileSize(file);
+      checkFileDuplicity(file);
+      checkQueueSize();
+      addToQueue(getUpdatedFileName(file.name), file);
+      console.log('fileProcessHandler after', fileQueue.length, fileQueue2.length);
+    } catch (error) {
+      if (onError) {
+        onError(error);
+      } else {
+        console.warn(error);
+      }
+    }
+  };
+
+  const onDragOverHandler = (event: Event) => {
```

Comment:
If you want to split this hook, what will be left here? Here, all other links to fileQueue are resolved, including all error messages, and I return the status during dragging. According to your proposal, I would have to split the handlers so that they are available for both hooks, which seems unnecessary to me.

Reply:
I'm just wondering why we should do this? Is there something wrong with this hook, or something that doesn't belong there?

Reply:
done

Reply:
To answer your why: [The separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). Also you can read https://dev.to/tamerlang/separation-of-concerns-the-simple-way-4jp2.

The simple answer is the clean code and the code in small pieces that do only one thing and one thing only. See https://ncrcoe.gitbooks.io/java-for-small-teams/content/style/400_methods_should_do_one_thing.html
This is the same for methods, functions, and classes. Decoupling code into smaller pieces that takes care of only one job makes the code more readable, maintainable, and testable.